components/Tag.js

```diff
@@ -1,17 +1,26 @@
 import Link from "../components/Link";
 import { abbreviateNumber } from "../services/utils/abbreviateNumbers";
+import { useRouter } from "next/router";
 
-export default function Tag({ name, total }) {
+export default function Tag({ name, total, path, selected }) {
+  const router = useRouter();
+  const { search } = router.query;
   return (
     <div className="relative">
       <Link
-        href={`/search?search=${name}`}
-        className="flex flex-row p-1 m-2 rounded-lg text-sm font-mono border-2 cursor-pointer shadow-none hover:border-orange-600"
+        href={path && path ==='/search'?
+        (!selected?
+        (search === ''?
```

Comment:
Here is should be `!search` instead of `search===' '` as on the initial page load the search is `undefined` and you are checking for `""` so it turned out to be false 
hence `/search?search=${search}and${name}` is called but `search is undefined`

that's why the search goes for `undefined and tag`

Reply:
Yeah I have solved that issue I am just looking at something else now and then I will make a commit